doc/user/tutorials/making_your_first_app.rst

```diff
@@ -0,0 +1,510 @@
+        @staticmethod
+        @plugins.HOOKIMPL
+        def exposeInterfaces(cs):
+            kernels = [
+                interfaces.InterfaceInfo(ORDER.FLUX, fluxSolver.FluxInterface, {}),
+                interfaces.InterfaceInfo(ORDER.THERMAL_HYDRAULICS, thermalSolver.ThermalInterface, {}),
+            ]
+            return kernels
+
+Creating the physics kernels
+============================
+So far we have basically been weaving an administrative thread to tell ARMI about the code
+we want to run. Now we finally get to write the guts of the code that actually does
+something. In your real app, this code will run your own industrial or research code, or
+perform your own methodology.  Here we just have it make up dummy values representing flux
+and temperatures.
+
+Making the dummy flux kernel
+----------------------------
+In a previous tutorial, we made a function that sets a dummy flux to all parts of the core
+based on a radial distance from the origin. Here we will re-use that code but package it
+more formally so that ARMI can actually run it for us from a user perspective.
+
+The interface is responsible largely for scheduling activities to run at various time
+points. For a flux calculation, we want it to compute at every single time node, so we use
+the :py:meth:`armi.interfaces.Interface.interactEveryNode` hook.
+
+These interaction hooks can call arbitrarily complex code. The code could, for example:
+
+* Run an external executable locally
+* Submit an external code to a cloud HPC and wait for it to complete
+* Run an internal physics tool
+
+Here it just a tiny bit of math locally.
+
+.. code-block:: python
+    :caption: ``myapp/fluxSolver.py``
+
+    import os
+
+    import numpy as np
+
+    from armi import runLog
+    from armi import interfaces
+    from armi.physics import neutronics
+
+
+    class FluxInterface(interfaces.Interface):
+        name = "dummyFlux"
+
+        def interactEveryNode(self, cycle=None, timeNode=None):
+            runLog.info("Computing neutron flux and power.")
+            setFakePower(self.r.core)
+
+
+    def setFakePower(core):
+        midplane = core[0].getHeight()/2.0
+        center = np.array([0,0,midplane])
+        peakPower = 1e6
+        mgFluxBase = np.arange(5)
+        for a in core:
+            for b in a:
+                vol = b.getVolume()
+                coords = b.spatialLocator.getGlobalCoordinates()
+                r = np.linalg.norm(abs(coords-center))
+                fuelFlag = 10 if b.isFuel() else 1.0
+                b.p.power = peakPower / r**2 * fuelFlag
+                b.p.pdens = b.p.power/vol
+                b.p.mgFlux = mgFluxBase*b.p.pdens
+
+
+Making the thermal/hydraulics kernel
+------------------------------------------
+Since we told the ARMI plugin to schedule the flux solver before thermal/hydraulics solver
+via the ``ORDER`` attribute, we can depend on there being up-to-date block-level ``power``
```

Comment:
`ORDER` probably deserves a little more discussion. That the framework ships with a pre-supposed ORDER is a little surprising, so it might be worth mentioning that what is already in Interfaces is a good starting point, and really any ORDER can be used; Interfaces will just be sorted by order by the Operator.